External link source for 'k8s.io/client-go/tools/clientcmd/api.ExecConfig' is not found

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

When running `make generate-apiref` we have a warning:

```
E0130 09:47:04.566212   30664 types.go:298] External link source for 'k8s.io/client-go/tools/clientcmd/api.ExecConfig' is not found.
```

**What you expected to happen**:

No warnings

**How to reproduce it (as minimally and precisely as possible)**:

Run:

```
make generate-apiref`
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Kueue version (use `git describe --tags --dirty --always`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

Hmm, I thought my build failed on that one:

https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/8913/pull-kueue-verify-main/2017242111480434688

PTAL:

```
Generating lister code for 5 targets
No depcheck issue
Generating informer code for 5 targets
I0130 14:29:32.561875   30912 main.go:325] Output written to /home/prow/go/src/sigs.k8s.io/kueue/site/content/en/docs/reference/kueue-config.v1beta1.md
I0130 14:29:32.561911   30912 main.go:142] Parsing go packages in sigs.k8s.io/kueue/apis/config/v1beta2
E0130 14:30:16.594688   30912 types.go:298] External link source for 'k8s.io/client-go/tools/clientcmd/api.ExecConfig' is not found.
I0130 14:30:16.603525   30912 main.go:325] Output written to /home/prow/go/src/sigs.k8s.io/kueue/site/content/en/docs/reference/kueue-config.v1beta2.md
make[1]: Leaving directory '/home/prow/go/src/sigs.k8s.io/kueue'
make: *** [Makefile-verify.mk:56: verify] Error 2
```
but maybe indeed the underlying issue was different

--- a/hack/genref/config.yaml
+++ b/hack/genref/config.yaml
@@ -35,6 +35,8 @@ externalPackages:
     target: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#{{- lower .TypeIdentifier -}}-{{- arrIndex .PackageSegments -1 -}}-{{- arrIndex .PackageSegments -2 -}}
   - match: ^k8s\.io/apimachinery/pkg/api/resource\.Quantity$
     target: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity
+  - match: ^k8s\.io/client-go/tools/clientcmd/api\.
+    target: https://pkg.go.dev/k8s.io/client-go/tools/clientcmd/api#{{- .TypeIdentifier -}}
   - match: ^k8s.io/component-base/config/v1alpha1.
     target: https://pkg.go.dev/k8s.io/component-base/config/v1alpha1#{{- .TypeIdentifier -}}
   - match: ^time\.Duration$

--- a/site/content/en/docs/reference/kueue-config.v1beta2.md
+++ b/site/content/en/docs/reference/kueue-config.v1beta2.md
@@ -136,7 +136,7 @@ using the ClusterProfile API.</p>
 </td>
 </tr>
 <tr><td><code>execConfig</code> <B>[Required]</B><br/>
-<code>k8s.io/client-go/tools/clientcmd/api.ExecConfig</code>
+<a href="https://pkg.go.dev/k8s.io/client-go/tools/clientcmd/api#ExecConfig"><code>k8s.io/client-go/tools/clientcmd/api.ExecConfig</code></a>
 </td>
 <td>
    <p>ExecConfig is the exec configuration to obtain credentials.</p>

